test/error-address-preservation.spec.ts

```diff
@@ -0,0 +1,12 @@
+import {ErrorType, HyperFormula} from '../src'
+import {adr, detailedErrorWithOrigin} from './testUtils'
+
+describe('Address preservation.', () => {
+  it('Should work in the basic case.', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=NA()', '=A1']
+    ])
+    expect(engine.getCellValue(adr('A1'))).toEqualError(detailedErrorWithOrigin(ErrorType.NA, 'Sheet1!A1'))
+    expect(engine.getCellValue(adr('B1'))).toEqualError(detailedErrorWithOrigin(ErrorType.NA, 'Sheet1!A1'))
+  })
+})
```

Comment:
Please add a few more test cases. 

Does it work with Named Expressions (they are in `-1` worksheet)? 
Does it work with a function, nested function? 
With operators? 
Between two worksheets? 
CRUD operations? (I guess that removing sheet/column will create a new #REF error anyway)

Reply:
Added some tests. This logic is orthogonal to CRUDs, so any fault here would be fault in CRUDs, and should be tested with CRUDs.

Reply:
Yeah, true 👍 Thanks for the tests. I've seen that we support all those scenarios, just wanted avoid regressions in the future.